client/network-gossip/src/lib.rs

```diff
@@ -97,8 +103,26 @@ impl<B: BlockT, H: ExHashT> Network<B> for Arc<NetworkService<B, H>> {
 		NetworkService::report_peer(self, peer_id, reputation);
 	}
 
-	fn disconnect_peer(&self, who: PeerId) {
-		NetworkService::disconnect_peer(self, who)
+	fn add_set_reserved(&self, who: PeerId, protocol: Cow<'static, str>) {
+		let addr = iter::once(multiaddr::Protocol::P2p(who.into()))
+			.collect::<multiaddr::Multiaddr>();
+		let result = NetworkService::add_to_peers_set(self, protocol, iter::once(addr).collect());
```

Comment:
```suggestion
		let result = NetworkService::add_peers_set_reservedself, protocol, iter::once(addr).collect());
```

Should the node not be added as a reserved node?

Reply:
Indeed. I changed that here: https://github.com/paritytech/substrate/pull/7700#issuecomment-741768066
I'm a bit unsure what to do.